sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/EventGridExtensionConfigProvider.cs

```diff
@@ -174,6 +183,50 @@ private async Task<HttpResponseMessage> ProcessEventsAsync(JArray events, string
             return new HttpResponseMessage(HttpStatusCode.Accepted);
         }
 
+        private async Task<FunctionResult> ExecuteWithTracingAsync(string functionName, TriggeredFunctionData triggerData)
+        {
+            using DiagnosticScope scope = _diagnosticScopeFactory.CreateScope(functionName, DiagnosticScope.ActivityKind.Consumer);
```

Comment:
Is it correct to name the scope after the function name?

Reply:
I think we don't do it for EH.

Reply:
I used function name here because that functions will do on the telemetry.
BTW, Functions ignore trigger activity names and use function names for EH SDK Activities (and all others) anyway.

https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs#L517. 

https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Initializers/WebJobsTelemetryInitializer.cs#L68

Otel also wants to see the function name there.

> Span name should be set to the function name being executed.

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/faas.md#general-attributes


What do you think? What's the concern about naming?

Reply:
>What's the concern about naming?

mostly consistency. 

I'm ok using the function name as a span name but if we decide on it we need to check SB and EH

Reply:
i think H/SB extension use instrumentation from SDK and therefore we don't need and can't do anything about naming.

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.cs

Reply:
We can try and set Activity.Name before running the function code.

Reply:
will do - [DisplayName](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L134) is settable

Reply:
PS: Activity name (`OperationName`) is not settable, but `DisplayName` should be the one used by OTel.

Reply:
You don't have to do it as part of this PR. We just need to decide what are we doing form now on :)

Reply:
sure, I created https://github.com/Azure/azure-sdk-for-net/issues/25996 - i think it will require a bit of reflection magic, so I'd rather do it for EH/SB/EG is a separate PR and leave this one as is if you don't mind.

Reply:
Ok, I changed Activity name to `EventGrid.Process` in this PR. I've tried to also set `DisplayName` and think it should rather be set in Function runtime - more info here https://github.com/Azure/azure-sdk-for-net/issues/25996#issuecomment-997142406